### PR TITLE
Added options validator as discussed on #87

### PR DESCRIPTION
**Pull request by galvesribeiro**

### CODE REVIEW

**Comment by galvesribeiro** [comment]
cc: @stephenlautier

File: src/SignalR.Orleans/HostingExtensions.cs [hunk 1]
```diff
@@ -47,27 +36,38 @@ public static ISiloHostBuilder UseSignalR(this ISiloHostBuilder builder, Action<
...
     }
+
+    internal class SignalRConfigurationValidator : IConfigurationValidator
+    {
+        private readonly IServiceProvider _sp;
+        private readonly ILogger _logger;
+
+        public SignalRConfigurationValidator(IServiceProvider serviceProvider)
+        {
+            this._logger = serviceProvider.GetRequiredService<ILoggerFactory>().CreateLogger<SignalRConfigurationValidator>();
```
Comment (stephenlautier) [inline src/SignalR.Orleans/HostingExtensions.cs]: i guess you can change this to `ILogger<SignalRConfigurationValidator> logger` in the ctor

File: src/SignalR.Orleans/HostingExtensions.cs [hunk 2]
```diff
@@ -14,23 +19,7 @@ public static ISiloBuilder UseSignalR(this ISiloBuilder builder, Action<SignalrO
...
-
-            try
-            {
-                builder.AddMemoryGrainStorage(Constants.STORAGE_PROVIDER);
-            }
-            catch
-            {
-                /** Grain storage provider was already added. Do nothing. **/
-            }
+            builder.ConfigureServices(services => services.AddSingleton<IConfigurationValidator,SignalRConfigurationValidator>());
```
Comment (stephenlautier) [inline src/SignalR.Orleans/HostingExtensions.cs]: I'm going to leave this here (unrelated to this line) but i think we had missed another one (even in my PR) `UseSignalR` in https://github.com/OrleansContrib/SignalR.Orleans/blob/master/src/SignalR.Orleans/Extensions.cs#L33

File: src/SignalR.Orleans/HostingExtensions.cs [hunk 3]
```diff
@@ -14,23 +19,7 @@ public static ISiloBuilder UseSignalR(this ISiloBuilder builder, Action<SignalrO
...
-                builder.AddMemoryGrainStorage(Constants.PUBSUB_PROVIDER);
-            }
-            catch
-            {
-                /** PubSubStore was already added. Do nothing. **/
-            }
-
-            try
-            {
-                builder.AddMemoryGrainStorage(Constants.STORAGE_PROVIDER);
```
Comment (stephenlautier) [inline src/SignalR.Orleans/HostingExtensions.cs]: should you also remove this? -- i believe if you remove this is a bit "breaking" -- or perhaps  add doc to also register this (as it has custom storage name unlike pubsub)

File: src/SignalR.Orleans/HostingExtensions.cs (hunk 2 above)
Comment (galvesribeiro) [reply src/SignalR.Orleans/HostingExtensions.cs]: Oh! Yeah... the other storge provider... Yeah, I should validate that as well. Thanks!

File: src/SignalR.Orleans/HostingExtensions.cs (hunk 3 above)
Comment (galvesribeiro) [reply src/SignalR.Orleans/HostingExtensions.cs]: Yeah... I should add that but as it is internal, the user should only tell the name on the options and we register that ourselves...

Comment (stephenlautier) [reply src/SignalR.Orleans/HostingExtensions.cs]: Yes, but either internal we need to register or the consumer needs too if no one does, there won't be any provider registered I guess

Also, just a note - even if its internal the consumer should be able to register it to use a custom grain storage provider

Comment (galvesribeiro) [reply src/SignalR.Orleans/HostingExtensions.cs]: I had a second thought and brought back the old code to register the provider.

I need to think better about this change. Perhaps we should change the grain to use the Storage APIs which are based on Facets subsystem in Orleans. 

For now, lets keep as is. Need to research more and will push another PR. Thnanks!

**Pull request merged**